chains/evm/contracts/onRamp/OnRamp.sol

```diff
@@ -452,53 +457,64 @@ contract OnRamp is IEVM2AnyOnRampClient, ITypeAndVersion, Ownable2StepMsgSender
   /// @notice Parses and validates extra arguments, applying defaults from destination chain configuration.
   /// The function ensures all messages have the required CCVs and executor needed for processing,
   /// even when users don't explicitly specify them.
+  /// @param destChainSelector The destination chain selector.
   /// @param destChainConfig Configuration for the destination chain including default values.
   /// @param extraArgs User-provided extra arguments in either V3 or legacy format.
+  /// @param isTokenTransferWithoutData Indicates if the message has no data but includes a token transfer. This is meant to
+  /// signal token-only transfers to avoid adding default CCVs when not needed.
   /// @return resolvedArgs Complete EVMExtraArgsV3 struct with all defaults applied.
   function _parseExtraArgsWithDefaults(
     uint64 destChainSelector,
     DestChainConfig memory destChainConfig,
```

Comment:
If you make this storage instead of memory, it avoids copying the entire struct, though this only saves a few warm SLOADS because you later read from the slots that are not used in this function. It also requires refactoring the OnRampHelper to use storage instead of a passed in memory struct. 

So food for thought though if the test refactor is annoying its not saving a ton.